### DISCUSSION TITLE
Disable eglot when working with remote files (Tramp over SSH)

### DISCUSSION THREAD

**Discussion by quimm2003** [joaotavora/eglot#1298] [Q&A] [answered]
Hi there,

I need to know how to disable eglot when I open a remote file using SSH.

That's because I work with servers where I can not install any feature, including language servers. So, eglot is trying to open the appropriate language server on the remote server, and it slows the load of the file.

Can this be done using eglot-stay-out-of?

Thanks in advance.

**Comment by joaotavora** [comment]
To disable, just don't enable.  Should be pretty simple.

**Reply by quimm2003** [reply]
It's too simple. I have eglot enabled globally for python, c, and sh. I work with local files and remote files at same time. I don't want to disable eglot on local files but I do want to disable it when working with remote files.

Wouldn't it be useful to have a variable or a function to disable eglot automatically when working with remote files? If eglot is enabled by default for remote files, it would be nice to be able to disable it whenever you want.

**Comment by joaotavora** [comment]
> I have eglot enabled globally for python, c, and sh. 

I don't understand what you are  taking about I'm afraid. Eglot is not a global minor mode.

**Reply by joaotavora** [reply]
You didn't provide any code. Are you using `eglot-ensure` in a major mode hook?

**Reply by quimm2003** [reply]
Ups, sorry. This is my eglot initialization code:

```

(use-package eglot)

(with-eval-after-load 'eglot
  (add-to-list 'eglot-server-programs
               '((c-mode c++-mode)
                 . ("clangd"
                    "-j=8"
                    "--log=error"
                    "--background-index"
                    "--clang-tidy"
                    "--cross-file-rename"
                    "--completion-style=detailed"
                    "--pch-storage=memory"
                    "--header-insertion=never"
                    "--header-insertion-decorators=0"))))

;; disable flymake
(add-hook 'eglot--managed-mode-hook (lambda () (flymake-mode -1)))

(add-hook 'python-mode-hook 'eglot-ensure)
(add-hook 'sh-mode-hook 'eglot-ensure)
(add-hook 'c-mode-hook 'eglot-ensure)

(use-package flycheck-eglot)

(with-eval-after-load 'eglot
     (global-flycheck-eglot-mode 1))

```

**Reply by joaotavora** [reply, marked as answer]
OK, then your fix is very simple.  Delete the lines that contain `eglot-ensure` and start getting used to `M-x eglot` when you need to use Eglot.  `eglot-ensure` is suppoted, but [discouraged](https://joaotavora.github.io/eglot/#Starting-Eglot).

```
Note that this variant of starting an Eglot session is non-interactive, so it should be used only when you are confident that Eglot can be started reliably for any file which may be visited with the major-mode in question.
```

Alternatively, learn to write a lisp snippet that checks if a file is remote before calling `eglot-ensure`.

**Reply by quimm2003** [reply]
You're right. Thanks Joao.
